hacspec/src/traits.rs

```diff
@@ -65,7 +65,45 @@ pub trait SeqTrait<T: Copy> : Index<usize, Output=T> + IndexMut<usize, Output=T>
     }
 }
 
-pub trait SecretInteger {}
+// FIXME: rename
+/// This trait extends the `Numeric` trait and is implemented by all integer
+/// types. It offers bit manipulation, instantiation from literal, and convenient
+/// constants.
+pub trait Integer: Numeric {
+    const NUM_BITS: u32;
+    const ZERO: Self;
+    const ONE: Self;
+    const TWO: Self;
+
+    /// Get an integer with value `val`.
+    fn from_literal(val: u128) -> Self;
+
+    /// Get bit `i` of this integer.
+    #[inline]
+    fn get_bit(self, i: u32) -> Self {
+        (self >> i) & Self::ONE
+    }
+
+    /// Set bit `i` of this integer to `b` and return the result.
+    /// Bit `b` has to be `0` or `1`.
+    #[inline]
+    fn set_bit(self, b: Self, i: u32) -> Self {
```

Comment:
Why not take `b : bool` instead of `b: Self` ?

Reply:
I think there's no consensus on whether `bool <-> int` conversions are constant time. `Self` should be the safer choice.

Reply:
ok !